From 4833b9c66251dc7bf0aa6d75292de633093ffea5 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@gnome.org>
Date: Mon, 30 Apr 2012 20:17:14 +0000
Subject: Unavailable entries were displayed incorrectly for signatures

 * Only load unknown keys for ones that we can't find locally

https://bugzilla.gnome.org/show_bug.cgi?id=675015
---
diff --git a/pgp/seahorse-pgp-backend.c b/pgp/seahorse-pgp-backend.c
index b473a8a..36df01f 100644
--- a/pgp/seahorse-pgp-backend.c
+++ b/pgp/seahorse-pgp-backend.c
@@ -701,20 +701,20 @@ seahorse_pgp_backend_discover_keys (SeahorsePgpBackend *self,
 		robjects = g_list_prepend (robjects, key);
 	}
 
-	g_ptr_array_add (todiscover, NULL);
-
-	/* Start a discover process on all todiscover */
-	if (todiscover != NULL &&
-	    g_settings_get_boolean (seahorse_context_settings (NULL), "server-auto-retrieve")) {
-		seahorse_pgp_backend_retrieve_async (self, (const gchar **)todiscover->pdata,
-		                                     SEAHORSE_PLACE (self->keyring),
-		                                     cancellable, NULL, NULL);
-	}
-
-	/* Add unknown objects for all these */
-	for (i = 0; keyids[i] != NULL; i++) {
-		object = seahorse_unknown_source_add_object (self->unknown, keyids[i], cancellable);
-		robjects = g_list_prepend (robjects, object);
+	if (todiscover->len > 0) {
+		g_ptr_array_add (todiscover, NULL);
+		keyids = (const gchar **)todiscover->pdata;
+
+		/* Start a discover process on all todiscover */
+		if (g_settings_get_boolean (seahorse_context_settings (NULL), "server-auto-retrieve"))
+			seahorse_pgp_backend_retrieve_async (self, keyids, SEAHORSE_PLACE (self->keyring),
+			                                     cancellable, NULL, NULL);
+
+		/* Add unknown objects for all these */
+		for (i = 0; keyids[i] != NULL; i++) {
+			object = seahorse_unknown_source_add_object (self->unknown, keyids[i], cancellable);
+			robjects = g_list_prepend (robjects, object);
+		}
 	}
 
 	g_ptr_array_free (todiscover, TRUE);
--
cgit v0.9.0.2
